azurerm/resource_arm_automation_schedule.go

```diff
@@ -107,14 +108,78 @@ func resourceArmAutomationSchedule() *schema.Resource {
 				//todo figure out how to validate this properly
 			},
 
-			//todo missing properties: week_days, month_days, month_week_day from advanced automation section
+			"advanced_schedule": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"week_days": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type:         schema.TypeString,
+								ValidateFunc: validateScheduleDay(),
+							},
+							Set: set.HashStringIgnoreCase,
+						},
+						"month_days": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type:         schema.TypeInt,
+								ValidateFunc: validate.IntBetweenAndNot(-1, 31, 0),
+							},
+							Set: set.HashInt,
+						},
+
+						"monthly_occurrence": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"day": {
+										Type:             schema.TypeString,
+										Required:         true,
+										DiffSuppressFunc: suppress.CaseDifference,
+										ValidateFunc:     validateScheduleDay(),
```

Comment:
can we in-line this validation method for the moment? I have a feeling we may end up extracting out a `listOfWeekDays` method in time